### PR TITLE
Migrate the text command system to the "new" command system.

### PR DESCRIPTION
**Pull request by Kodehawa**
NewCommand (renamed to TextCommand on this PR) has been in place for a long time, but text commands never got migrated to it.

This migrates all text commands to TextCommand, and removes the old Command/TreeCommand system in place since 2017.

This is still WIP.

### CODE REVIEW

File: src/main/java/net/kodehawa/mantarobot/commands/HelpCmd.java [hunk 1]
```diff
@@ -564,9 +518,35 @@ private void buildHelp(Context ctx, CommandCategory category) {
...
+    @Alias("skip") @Alias("stop") @Alias("ns")
+    @Alias("volume") @Alias("forceplay") @Alias("lyrics")
+    @Alias("playnow") @Alias("rewind") @Alias("forward")
+    @Alias("restartsong") @Alias("removetrack") @Alias("move")
+    @Alias("cast") @Alias("salvage") @Alias("repeat")
+    @Alias("repair") @Alias("marry") @Alias("divorce")
+    @Alias("mute") @Alias("unmute") @Alias("remindme")
+    @Alias("game") @Alias("trivia") @Alias("love")
+    @Category(CommandCategory.HIDDEN)
+    public static class Slash extends TextCommand {
```
Comment (MrLar) [inline src/main/java/net/kodehawa/mantarobot/commands/HelpCmd.java:549]: In all 100% serious do we really still need this?

File: src/main/java/net/kodehawa/mantarobot/commands/UtilsCmds.java [hunk 2]
```diff
@@ -291,64 +289,103 @@ private static List<ReminderObject> getReminders(List<String> reminders) {
...
+                ctx.send(EmoteReference.OK + "**For a feature overview, check:** https://mantaro.site/features.html");
+            }
+        }
+
+        public static class Birthday extends TextCommand {
+            @Override
+            protected void process(TextContext ctx) {
+                ctx.send(EmoteReference.OK + "**For a guide on the birthday system, please visit:** https://www.mantaro.site/mantaro-wiki/guides/birthday-announcer");
+            }
+        }
```
Comment (MrLar) [inline src/main/java/net/kodehawa/mantarobot/commands/UtilsCmds.java:389]: Realistically do all of these need to be separate classes? We could just have the argument resolved in a single class. Might need some hackery for help context tho.

File: src/main/java/net/kodehawa/mantarobot/commands/ActionCmds.java [hunk 3]
```diff
@@ -338,129 +381,216 @@ public Cuddle() {
...
+    public static class Nom extends ImageCmd {
+        public Nom() {
+            super("*nom nom*", "nom", "commands.action.nom");
+        }
+    }
+
+    public static class Facedesk extends ImageCmd {
+        public Facedesk() {
+            super("When it's just too much to handle.", "banghead", "commands.action.facedesk", true);
+        }
```
Comment (MrLar) [inline src/main/java/net/kodehawa/mantarobot/commands/ActionCmds.java:594]: We should consider making a Hybrid class for these types of commands that handles both text and slash. If you look more closely you can see that each and every single one of these has the same exact class structure, they just rely on 2 different Contexts - which we can probably abstract away.

File: src/main/java/net/kodehawa/mantarobot/core/command/TextContext.java [hunk 4]
```diff
@@ -496,6 +552,20 @@ public User getAuthor() {
         return getMessage().getAuthor();
     }
 
+    public Marriage getMarriage(MongoUser userData) {
+        return MantaroData.db().getMarriage(userData.getMarriageId());
+    }
```
Comment (MrLar) [inline src/main/java/net/kodehawa/mantarobot/core/command/TextContext.java:557]: Both TextContext and BaseInteractionContext now have this method and both implementations have the exact same body. This should really be a default method in IContext.

File: src/main/java/net/kodehawa/mantarobot/core/command/TextContext.java [hunk 5]
```diff
@@ -487,6 +527,22 @@ public Player getPlayer(String id) {
...
+
+    public PlayerStats getPlayerStats(User user) {
+        return managedDatabase.getPlayerStats(user);
+    }
+
+    public PlayerStats getPlayerStats(Member member) {
+        return managedDatabase.getPlayerStats(member);
+    }
+
```
Comment (MrLar) [inline src/main/java/net/kodehawa/mantarobot/core/command/TextContext.java:545]: Both TextContext and BaseInteractionContext now have these methods and both implementations have the exact same bodies. These should really be a default methods in IContext.

File: src/main/java/net/kodehawa/mantarobot/core/command/TextContext.java [hunk 6]
```diff
@@ -506,9 +576,78 @@ public UtilsContext getUtilsContext() {
...
     public I18nContext getLanguageContext() {
         return i18n;
     }
+
+    public void setLanguageContext(I18nContext languageContext) {
+        this.i18n = languageContext;
+    }
+
+    public I18nContext getGuildLanguageContext() {
+        return new I18nContext(getDBGuild(), null);
```
Comment (MrLar) [inline src/main/java/net/kodehawa/mantarobot/core/command/TextContext.java:609]: Same as the other 2 comments about default in IContext.

File: src/main/java/net/kodehawa/mantarobot/core/command/TextContext.java [hunk 7]
```diff
@@ -506,9 +576,78 @@ public UtilsContext getUtilsContext() {
...
+    }
+
+    public Color getMemberColor(Member member) {
+        return member.getColor() == null ? Color.PINK : member.getColor();
+    }
+
+    public Color getMemberColor() {
+        return getMemberColor(getMember());
+    }
```
Comment (MrLar) [inline src/main/java/net/kodehawa/mantarobot/core/command/TextContext.java:618]: Same as the other 2 comments about default in IContext.

File: src/main/java/net/kodehawa/mantarobot/core/command/TextContext.java [hunk 8]
```diff
@@ -506,9 +576,78 @@ public UtilsContext getUtilsContext() {
...
+        return content.map(String::trim).orElse("");
+    }
+
+    public boolean isChannelNSFW() {
+        if (getChannel() instanceof IAgeRestrictedChannel txtChannel) {
+            return txtChannel.isNSFW();
+        }
+
+        return true;
+    }
```
Comment (MrLar) [inline src/main/java/net/kodehawa/mantarobot/core/command/TextContext.java:648]: Same as the other 2 comments about default in IContext. This should probably also not default to true?

**Review by MrLar — CHANGES_REQUESTED** [review]

File: src/main/java/net/kodehawa/mantarobot/commands/HelpCmd.java (hunk 1 above)
Comment (Kodehawa) [reply src/main/java/net/kodehawa/mantarobot/commands/HelpCmd.java]: Removed, it's not

**Pull request merged**